packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1540,6 +1570,8 @@ export class KeyringController extends BaseController<
    * when initializing the controller
    */
   async #addQRKeyring(): Promise<QRKeyring> {
+    this.#assertControllerMutexIsLocked();
```

Comment:
The main reason for that is to avoid the addition of methods in the future that could unintentionally call these methods internally without locking: if it's true that _all_ KeyringController mutable operations must be atomic, then should be fair to assert that all mutable methods (including `#` methods) must be behind the mutex.